stash/src/repository/TransactionRepository.ts

```diff
@@ -1,24 +1,38 @@
 import { Client } from 'redis-om'
-import { transactionSchema } from '../model/Transaction.js'
+import { depositSchema } from '../model/Deposit.js'
+import { withdrawalSchema } from '../model/Withdrawal.js'
 import { getTimeseriesUrl } from '../connector/RedisConnector.js'
 import logger from '../util/Logger.js'
 
 const client = new Client()
 await client.open(getTimeseriesUrl())
 
-const transactionRepository = client.fetchRepository(transactionSchema)
+const depositRepository = client.fetchRepository(depositSchema)
+const withdrawalRepository = client.fetchRepository(withdrawalSchema)
 
 try {
-  await transactionRepository.createIndex()
+  await depositRepository.createIndex()
 } catch (error) {
-  if (error.message.includes('Index already exists')) {
+  if (error.message.includes('Deposit index already exists')) {
     logger.log({
       level: 'info',
-      message: 'Index already exists, skipping creation.',
+      message: 'Deposit index already exists, skipping creation.',
+    })
+  } else {
```

Comment:
In general, the code can be written without using `else`. The good practice is to throw exception earlier, so in this case inverting the `if logic`